tests/test_calibrateImage.py

```diff
@@ -0,0 +1,396 @@
+        self.fluxes = np.array((6*noise*psf_area,
+                                12*noise*psf_area,
+                                45*noise*psf_area,
+                                150*noise*psf_area,
+                                400*noise*psf_area,
+                                1000*noise*psf_area))
+        self.centroids = np.array(((162, 22),
+                                   (25, 70),
+                                   (100, 160),
+                                   (50, 120),
+                                   (92, 35),
+                                   (175, 154)), dtype=np.float32)
+        for flux, centroid in zip(self.fluxes, self.centroids):
+            dataset.addSource(instFlux=flux, centroid=lsst.geom.Point2D(centroid[0], centroid[1]))
+
+        # Bright extended source in the center of the image: should not appear
+        # in any of the output catalogs.
+        center = lsst.geom.Point2D(100, 100)
+        shape = lsst.afw.geom.Quadrupole(8, 9, 3)
+        dataset.addSource(instFlux=500*noise*psf_area, centroid=center, shape=shape)
+
+        schema = dataset.makeMinimalSchema()
+        schema.addField("truth_flux", type=np.float64, doc="true flux", units="nJy")
+        schema.addField("truth_fluxErr", type=np.float64, doc="true fluxErr", units="nJy")
+
+        self.truth_exposure, self.truth_cat = dataset.realize(noise=noise, schema=dataset.makeMinimalSchema())
+        lsst.afw.table.updateSourceCoords(self.truth_exposure.wcs, self.truth_cat)
+        # To make it look like a version=1 (nJy fluxes) refcat
+        self.truth_cat = self.truth_exposure.photoCalib.calibrateCatalog(self.truth_cat)
+        self.ref_loader = testUtils.MockReferenceObjectLoaderFromMemory([self.truth_cat])
+        metadata = lsst.daf.base.PropertyList()
+        metadata.set("REFCAT_FORMAT_VERSION", 1)
+        self.truth_cat.setMetadata(metadata)
+
+        # TODO: a cosmic ray (need to figure out how to insert a fake-CR)
+        # self.truth_exposure.image.array[10, 10] = 100000
+        # self.truth_exposure.variance.array[10, 10] = 100000/noise
+
+        # Copy the truth exposure, because CalibrateImage modifies the input.
+        # Post-ISR ccds only contain: initial WCS, VisitInfo, filter
+        self.exposure = afwImage.ExposureF(self.truth_exposure.maskedImage)
+        self.exposure.setWcs(self.truth_exposure.wcs)
+        self.exposure.info.setVisitInfo(self.truth_exposure.visitInfo)
+        # "truth" filter, to match the "truth" refcat.
+        self.exposure.setFilter(lsst.afw.image.FilterLabel(physical='truth', band="truth"))
+
+        # Test-specific configuration:
+        self.config = CalibrateImageTask.ConfigClass()
+        # We don't have many sources, so have to fit simpler models.
+        self.config.psf_detection.background.approxOrderX = 1
+        self.config.star_detection.background.approxOrderX = 1
+        # TODO: While debugging DM-32701, we're using PCA instead of psfex.
```

Comment:
Is this TODO a relic now?

Reply:
Unfortunately, no. DM-32701 fixed the "junk PSF produced" problem into a "raise error", but we still have the same problem of it thinking we don't have enough PSF sources (even when we should for the given fitting order). We don't have a followup ticket to explore that further, so I'm not sure what to say here.

Reply:
If this is just about the test, not any real data, let's just remove the TODO.

PSFEx _seeming inconsistent_ about how many stars it needs is something I'd consider a low-priority problem (to the point where I'm not sure a ticket or TODO is merited), because it's easy to imagine that happening because our thinking of its algorithm is an oversimplification and it genuinely has more degrees of freedom.

If the problem is that PSFEx just isn't robust enough on some important class of data, that's different.

Reply:
I think the problem is that we don't know which of those cases it is. This is basically the problem described on [DM-40001](https://jira.lsstcorp.org/browse/DM-40001). In this case, we can just use PCA since the tests don't really care, but I think we probably should investigate psfex's failure modes more fully here.